hnix-store-remote/src/System/Nix/Store/Remote/Logger.hs

```diff
@@ -34,12 +40,24 @@ processOutput = go decoder
           case ctrl of
             e@(Error _ _) -> return [e]
             Last -> return [Last]
+            Read n -> do
+              (mdata, _) <- get
+              case mdata of
+                Nothing -> throwError "No data to read provided"
+                Just part -> do
+                  -- XXX: we should check/assert part size against n of (Read n)
+                  sockPut $ putByteStringLen part
+                  clearData
+
+              next <- go decoder
+              return $ next
+
```

Comment:
This "async read" part of the protocol looks really weird. Any idea why it was implemented like that in the daemon ?